src/Symfony/Component/Translation/Test/ProviderFactoryTestCase.php

```diff
@@ -33,11 +34,11 @@
 abstract class ProviderFactoryTestCase extends TestCase
 {
     protected HttpClientInterface $client;
-    protected LoggerInterface $logger;
+    protected LoggerInterface|MockObject $logger;
     protected string $defaultLocale;
-    protected LoaderInterface $loader;
-    protected XliffFileDumper $xliffFileDumper;
-    protected TranslatorBagInterface $translatorBag;
+    protected LoaderInterface|MockObject $loader;
+    protected XliffFileDumper|MockObject $xliffFileDumper;
+    protected TranslatorBagInterface|MockObject $translatorBag;
```

Comment:
```suggestion
    protected LoaderInterface&MockObject $loader;
    protected XliffFileDumper&MockObject $xliffFileDumper;
    protected TranslatorBagInterface&MockObject $translatorBag;
```

Reply:
Nope, this change is on purpose. It's really an `|`. We do have a test case where a real instance of `XliffFileDumper` is returned. This made me realize that intersections with MockObject should quite often be unions, for the same reason - at least when inheritance is allowed.

Reply:
But in that case, the union would be too wide: It would allow virtually any mock object which is certainly not what we want.

Reply:
Yep, the union is still the best approximation to describe the type and provide ide autocompletion for mocks here. This is a test case, we don't care about type strictness.